tests/php/Core/Infrastructure/Security/ProviderConfiguration/WebSSO/Api/FindWebSSOConfiguration/FindWebSSOConfigurationControllerTest.php

```diff
@@ -0,0 +1,94 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Core\Infrastructure\Security\ProviderConfiguration\WebSSO\Api\FindWebSSOConfiguration;
+
+use Centreon\Domain\Contact\Contact;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Core\Application\Security\ProviderConfiguration\WebSSO\UseCase\FindWebSSOConfiguration\{
+    FindWebSSOConfiguration,
+    FindWebSSOConfigurationPresenterInterface
+};
+use Core\Infrastructure\Security\ProviderConfiguration\WebSSO\Api\FindWebSSOConfiguration\FindWebSSOConfigurationController;
+
+beforeEach(function () {
+    $this->useCase = $this->createMock(FindWebSSOConfiguration::class);
+    $this->presenter = $this->createMock(FindWebSSOConfigurationPresenterInterface::class);
+
+    $timezone = new \DateTimeZone('Europe/Paris');
+    $adminContact = (new Contact())
+        ->setId(1)
+        ->setName('admin')
+        ->setAdmin(true)
+        ->setTimezone($timezone);
+    $authorizationChecker = $this->createMock(AuthorizationCheckerInterface::class);
+    $authorizationChecker->expects($this->once())
+        ->method('isGranted')
+        ->willReturn(true);
+    $token = $this->createMock(TokenInterface::class);
+    $token->expects($this->any())
+        ->method('getUser')
+        ->willReturn($adminContact);
+    $tokenStorage = $this->createMock(TokenStorageInterface::class);
+    $tokenStorage->expects($this->any())
+        ->method('getToken')
+        ->willReturn($token);
+    $this->container = $this->createMock(ContainerInterface::class);
+    $this->container->expects($this->any())
+        ->method('has')
+        ->willReturn(true);
+    $this->container->expects($this->any())
+        ->method('get')
+        ->withConsecutive(
+            [$this->equalTo('security.authorization_checker')],
+            [$this->equalTo('parameter_bag')]
+        )
+        ->willReturnOnConsecutiveCalls(
+            $authorizationChecker,
+            new class () {
+                public function get(): string
+                {
+                    return __DIR__ . '/../../../../../';
+                }
+            }
+        );
+    $this->request = $this->createMock(Request::class);
+});
```

Comment:
we'll may need to create a custom helper to avoid to repeat this in each controller tests : https://pestphp.com/docs/helpers

Reply:
agreed